### ISSUE TITLE
Pruning extra revisions fail [3.7.45]: 

### ISSUE THREAD

**Issue by DavidKabelitz** [craftcms/cms#11507]
### What happened?

Hi,
we have about 300 errors a day at the queue manager. Always related to different entries but always the same error message.

Not sure what the problem is.

Thanks David

<img width="2031" alt="Bildschirmfoto 2022-06-27 um 13 40 04" src="https://user-images.githubusercontent.com/28010175/175933597-f79d41b6-f897-46ef-90ef-7644ca0be9bd.png">
0

### Craft CMS version

3.7.45

### Installed plugins and versions

Plugins in use: Cloudflare 1.1

**Comment by brandonkelly** [comment]
Try searching through `storage/logs/queue.log` (and any other `queue.log.X` files) for that error. Does it show up? If so please post the full stack trace that follows it.

**Comment by DavidKabelitz** [comment]
Yes it's there:

`2022-06-27 17:48:31 [-][-][-][info][craft\queue\QueueLogBehavior::beforeExec]  [1976457] Pruning extra revisions (attempt: 1, pid: 21062) - Started
2022-06-27 17:48:34 [-][-][-][error][craft\queue\QueueLogBehavior::afterError]  [1976457] Pruning extra revisions (attempt: 1, pid: 21062) - Error (time: 3.084s): Unable to acquire a lock for file "/var/www/craft/storage/logs/cloudflare.log".`

**Comment by DavidKabelitz** [comment]
or do you need the whole file? It's 8MB big

**Comment by brandonkelly** [comment]
I just need a stack trace. Is there one that follows the log you posted? If not try searching for other instances of the error message.

**Comment by DavidKabelitz** [comment]
just to inform you...after restarting all these errors again it works and they are gone. So looks like a timing issue ...don't know if its important but the cloudflare loaded log is after the queue error
`2022-06-27 18:06:44 [-][-][-][info][cloudflare] Cloudflare plugin loaded`

**Comment by angrybrad** [comment]
@DavidKabelitz do you have multiple queue runners processing the queue or just one? Is this in a load-balanced environment?

**Comment by DavidKabelitz** [comment]
@angrybrad yes we are running on kubernetes with multiple pods and also a load balancer

**Comment by angrybrad** [comment]
@DavidKabelitz How are the queue runner(s) set up? Using Craft's default web-based queue runner, so each pod ends up being a queue runner? Or is there a dedicated pod just for deamonizing `craft` to process the queue?

**Comment by DavidKabelitz** [comment]
Yes it's the default web based queue runner.
In that case we solved it and removed the queue runner cause we already had cron jobs running with schedules.

su www-data -c "/usr/local/bin/php /var/www/craft/craft queue/run"

So for a better / clean setup a dedicated pod for the queue would be the solution?
